services/business/checkers/no_service_checker.go

```diff
@@ -19,45 +19,57 @@ type NoServiceChecker struct {
 
 func (in NoServiceChecker) Check() models.IstioValidations {
 	validations := models.IstioValidations{}
+	validationsc := make(chan models.IstioValidations)
 
 	if in.IstioDetails == nil || in.ServiceList == nil {
 		return validations
 	}
 
-	tempVal := make([]models.IstioValidations, 4)
-	for i := range tempVal {
-		tempVal[i] = make(models.IstioValidations)
-	}
 	serviceNames := getServiceNames(in.ServiceList)
 
 	var wg sync.WaitGroup
 	wg.Add(4)
 
-	go runRouteRulesCheck(in.IstioDetails.RouteRules, in.Namespace, serviceNames, tempVal[0], &wg)
-	go runDestinationPoliciesCheck(in.IstioDetails.DestinationPolicies, in.Namespace, serviceNames, tempVal[1], &wg)
-	go runVirtualServicesCheck(in.IstioDetails.VirtualServices, in.Namespace, serviceNames, tempVal[2], &wg)
-	go runDestinationRulesCheck(in.IstioDetails.DestinationRules, in.Namespace, serviceNames, tempVal[3], &wg)
+	go runRouteRulesCheck(in.IstioDetails.RouteRules, in.Namespace, serviceNames, validationsc, &wg)
+	go runDestinationPoliciesCheck(in.IstioDetails.DestinationPolicies, in.Namespace, serviceNames, validationsc, &wg)
+	go runVirtualServicesCheck(in.IstioDetails.VirtualServices, in.Namespace, serviceNames, validationsc, &wg)
+	go runDestinationRulesCheck(in.IstioDetails.DestinationRules, in.Namespace, serviceNames, validationsc, &wg)
 
-	wg.Wait()
+	go func() {
+		wg.Wait()
+		// Closing the channel stop the range loop below
+		close(validationsc)
+	}()
 
-	for i := range tempVal {
-		validations.MergeValidations(tempVal[i])
+	for v := range validationsc {
+		validations.MergeValidations(v)
 	}
 	return validations
 }
 
-func runRouteRulesCheck(routeRules []kubernetes.IstioObject, namespace string, serviceNames []string, validations models.IstioValidations, wg *sync.WaitGroup) {
+func runRouteRulesCheck(routeRules []kubernetes.IstioObject, namespace string, serviceNames []string, validationsc chan models.IstioValidations, wg *sync.WaitGroup) {
 	defer wg.Done()
 	var rrWg sync.WaitGroup
 	rrWg.Add(len(routeRules))
+	rrvalidations := models.IstioValidations{}
+	rrvalidationsc := make(chan models.IstioValidations)
 
 	for _, routeRule := range routeRules {
-		go runRouteRuleCheck(routeRule, namespace, serviceNames, validations, &rrWg)
+		go runRouteRuleCheck(routeRule, namespace, serviceNames, rrvalidationsc, &rrWg)
```

Comment:
You should be able to pass `validationsc` down to `runRouteRuleCheck` (eg no need to create an intermediary `rrvalidationsc` channel) and keep the rest of the function unchanged. The same applies to `runDestinationPoliciesCheck`, `runVirtualServicesCheck` and `runDestinationRulesCheck`. Otherwise LGTM.